### PR TITLE
Plugin Search Part 1

### PR DESCRIPTION
**Pull request by codemonkey800**
## Description

This PR is part 1 of 2 PRs for adding plugin search. This PR includes minor refactors and additions accumulated while working on the search engine.

## Changes

- Running `yarn dev` will now run the mock server automatically
- It's now possible to run E2E tests using a specific screen size: `DEVICE=2xl yarn e2e`
- Index fixture data for testing
- Utilities:
  - `measureExecution()` for measuring runtime of a function using [window.performance.now()](https://developer.mozilla.org/en-US/docs/Web/API/Performance/now)
  - `Logger` for logging messages in a standardized way. Right now it's a glorified `console.log` wrapper, but it can be extended to other transports in the future
- Mock `window.location` in jest tests for router related tests
- `ColumnLayout` component for reusing the napari 3-column layout
- Allow support for custom page layouts using `getLayout()`

## Demos

Demos can be found in #35

### CODE REVIEW

File: client/src/components/common/ColumnLayout/ColumnLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+        // Padding
+        'p-6 md:p-14 2xl:px-0',
+
+        // Grid gap
+        '2xl:gap-14',
+
+        // Grid columns
+        '2xl:grid-cols-napari-2-col 3xl:grid-cols-napari-3-col',
+      )}
```
Comment (ehoops) [inline client/src/components/common/ColumnLayout/ColumnLayout.tsx:23]: The comments here are super helpful 🎉

File: client/src/pages/_app.tsx [hunk 2]
```diff
@@ -18,6 +18,10 @@ import { Hydrate } from 'react-query/hydration';
 import { Layout } from '@/components';
 import { MediaContextProvider } from '@/components/common/media';
 
+interface GetLayoutComponent {
+  getLayout?(page: ReactNode): ReactNode;
```
Comment (ehoops) [inline client/src/pages/_app.tsx:22]: I'm not familiar with this interface syntax.  What does the `?(page: ReactNode)` portion mean?  Is this related to javascript optional chaining?

Comment (ehoops) [reply client/src/pages/_app.tsx:22]: Is this an optional function?
```
getLayout?: (page: ReactNode) => ReactNode
```

File: client/src/pages/_app.tsx [hunk 3]
```diff
@@ -40,18 +44,20 @@ function ReactQueryProvider({ children, dehydratedState }: QueryProviderProps) {
 }
 
 export default function App({ Component, pageProps }: AppProps) {
+  // Render using custom layout if component exports one:
+  // https://adamwathan.me/2019/10/17/persistent-layout-patterns-in-nextjs/
```
Comment (ehoops) [inline client/src/pages/_app.tsx:48]: 🎉

File: client/src/components/common/ColumnLayout/ColumnLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import clsx from 'clsx';
+import { HTMLAttributes, ReactNode } from 'react';
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  children: ReactNode;
```
[the remark this answers is not in the document]
Comment (codemonkey800) [reply client/src/components/common/ColumnLayout/ColumnLayout.tsx:5]: It takes multiple (up to 3 nodes since it's a 3-column grid layout). Is there a better type to describe this? 🤔  `ReactNode` should cover both single and multiple children:

```ts
interface ReactNodeArray extends Array<ReactNode> {}
type ReactFragment = {} | ReactNodeArray;
type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```

File: client/src/utils/performance.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  duration: string;
+}
+
+/**
+ * Utility for measuring the execution duration of a function.
+ *
+ * @param fn Function to measure.
+ * @returns The result of the function, if any.
+ */
+export function measureExecution<R>(
```
Comment (justinelarsen) [inline client/src/utils/performance.ts:11]: Is this used for debugging? I don't see it used anywhere so just wondering how it's going to be used.

**Review by justinelarsen — APPROVED** [review]
LGTM Nice work @codemonkey800 !
